### PR TITLE
curvefs/mds: change createfs interface to an idempotent one

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by SeanHai** [comment]
recheck

**Comment by SeanHai** [comment]
recheck

**Comment by SeanHai** [comment]
recheck

**Comment by SeanHai** [comment]
recheck

File: curvefs/src/mds/fs_manager.cpp [hunk 1]
```diff
@@ -57,20 +57,28 @@ FSStatusCode FsManager::CreateFs(const std::string& fsName,
     // 1. query fs
     // TODO(cw123): if fs status is FsStatus::New, here need more consideration
     if (fsStorage_->Exist(fsName)) {
-        if (!IsPreviousCreateUnComplete(fsName, fsType, blockSize, detail)) {
+        int existRet = IsPreviousCreateUnCompleteOrExist(fsName, fsType,
```
Comment (wu-hanqing) [inline curvefs/src/mds/fs_manager.cpp:60]: it's better to rename it to `IsExactlySameOrCreateUnComplete`

File: curvefs/src/mds/fs_manager.h [hunk 2]
```diff
@@ -151,8 +151,9 @@ class FsManager {
                            FsInfo* fsInfo);
 
  private:
-    bool IsPreviousCreateUnComplete(const std::string& fsName, FSType fsType,
-                                    uint64_t blocksize, const FsDetail& detail);
+    // return 1: exist; 2: uncomplete; 0: neither
```
Comment (wu-hanqing) [inline curvefs/src/mds/fs_manager.h:154]: maybe `0: ExactlySame, 1: uncomplete, -1: neither` is better?

File: curvefs/src/mds/fs_manager.cpp (hunk 1 above)
Comment (SeanHai) [reply curvefs/src/mds/fs_manager.cpp:60]: > it's better to rename it to `IsExactlySameOrCreateUnComplete`

fixed

File: curvefs/src/mds/fs_manager.h (hunk 2 above)
Comment (SeanHai) [reply curvefs/src/mds/fs_manager.h:154]: > maybe `0: ExactlySame, 1: uncomplete, -1: neither` is better?

fixed

**Comment by SeanHai** [comment]
recheck

**Review by wu-hanqing — APPROVED** [review]

**Comment by SeanHai** [comment]
recheck

File: curvefs/src/mds/fs_manager.cpp [hunk 3]
```diff
@@ -57,20 +57,28 @@ FSStatusCode FsManager::CreateFs(const std::string& fsName,
     // 1. query fs
     // TODO(cw123): if fs status is FsStatus::New, here need more consideration
     if (fsStorage_->Exist(fsName)) {
-        if (!IsPreviousCreateUnComplete(fsName, fsType, blockSize, detail)) {
+        int existRet = IsExactlySameOrCreateUnComplete(fsName, fsType,
+                                                       blockSize, detail);
+        if (existRet == 0) {
             LOG(WARNING) << "CreateFs fail, fs exist, fsName = " << fsName
```
Comment (cw123) [inline curvefs/src/mds/fs_manager.cpp:63]: this log message should change

**Comment by SeanHai** [comment]
recheck

File: curvefs/src/mds/fs_manager.cpp (hunk 3 above)
Comment (SeanHai) [reply curvefs/src/mds/fs_manager.cpp]: > this log message should change

fixed

**Review by cw123 — APPROVED** [review]

**Pull request merged**